src/components/autocomplete/demoBasicUsage/index.html

```diff
@@ -18,7 +18,7 @@
         </md-item-template>
         <md-not-found>
           No states matching "{{ctrl.searchText}}" were found.
-          <a ng-click="ctrl.newState(ctrl.searchText)">Create a new one!</a>
+          <md-button ng-click="ctrl.newState(ctrl.searchText, $event)">Create a new one!</md-button>
```

Comment:
It is hard to focus on a PR that seems to be trying to do a lot of changes at once, especially when it includes changes like this line that seems unrelated (changing `<a>` to `<md-button>`)

I don't think I'll be able to comfortably be able to review a change like this given that it tries to make too many changes at once. I'd prefer to see a PR that works through a predetermined set of steps that brings us to our goal (remove use of globals), e.g. 
- Replace uses of document with $document
- Replace uses of console.log with md-dialog
- Replace uses of window with $window
- ...

If the PRs were separated as such, it would be much easier to parse and review. As it is now, it is too risky of a change to confidently LGTM

Reply:
that's just a doc change because they were using window.event.

I just swapped the a for a button because you can't have a `a` without an href, and I was editing that line anyway.

Reply:
We can try running it through presubmits once it is reviewed (@Splaktar - do you want to look at this). If it passes then it's alright with me

Reply:
Yeah, I'll review before adding the `merge ready` label and asking for it to be run through the presubmits.